apps/src/StudioApp.js

```diff
@@ -670,6 +671,12 @@ StudioApp.prototype.handleClearPuzzle = function (config) {
     this.editor.setValue(resetValue);
 
     annotationList.clearRuntimeAnnotations();
+  } else if (this.scratch) {
+    const workspace = Blockly.getMainWorkspace();
+    workspace.clear();
+
+    const dom = Blockly.Xml.textToDom(config.level.startBlocks);
+    Blockly.Xml.domToWorkspace(dom, workspace);
```

Comment:
Yep!  https://github.com/LLK/scratch-blocks is much closer to https://github.com/google/blockly than our fork of Blockly.